setup/services.py

```
@@ -172,7 +172,17 @@ def start_backend(host: str, port: int, debug: bool = False):
     import re
-    if not re.match(r'^[a-zA-Z0-9.-]+
+    if not re.match(r'^[a-zA-Z0-9.-]+$', host):
+        logger.error(f"Invalid host name: {host}")
+        return
+
+    # Use standard list format for subprocess.Popen to prevent command injection
+    cmd = [str(python_exe), "src/main.py", "--host", host, "--port", str(port)]
+
+    # Run the process
+    process = subprocess.Popen(cmd, cwd=ROOT_DIR)
+    process_manager.add_process(process, "Backend")
```

Flake8 (7.3.0)

[error] undefined name 'process_manager'

(F821)